js/renovation/ui/data_grid/data_grid.tsx

```diff
@@ -41,4 +54,343 @@ export class DataGrid extends JSXComponent(DataGridProps) {
+    return this.widget?.getCombinedFilter(returnDataField);
+  }
+
+  @Method()
+  getDataSource(): any /* DataSource */ {
+    return this.widget?.getDataSource();
+  }
+
+  @Method()
+  getKeyByRowIndex(rowIndex: number): any {
+    return this.widget?.getKeyByRowIndex(rowIndex);
+  }
+
+  @Method()
+  getRowElement(rowIndex: number): Element[] & JQuery | undefined {
+    return this.widget?.getRowElement(rowIndex);
+  }
+
+  @Method()
+  getRowIndexByKey(key: any | string | number): number {
+    return this.widget?.getRowIndexByKey(key);
+  }
+
+  @Method()
+  getScrollable(): any /* dxScrollable */ {
+    return this.widget?.getScrollable();
+  }
+
+  @Method()
+  getVisibleColumnIndex(id: number | string): number {
+    return this.widget?.getVisibleColumnIndex(id);
+  }
+
+  @Method()
+  hasEditData(): boolean {
+    return this.widget?.hasEditData();
+  }
+
+  @Method()
+  hideColumnChooser(): void {
+    return this.widget?.hideColumnChooser();
+  }
+
+  @Method()
+  isAdaptiveDetailRowExpanded(key: any): boolean {
+    return this.widget?.isAdaptiveDetailRowExpanded(key);
+  }
+
+  @Method()
+  isRowFocused(key: any): boolean {
+    return this.widget?.isRowFocused(key);
+  }
+
+  @Method()
+  isRowSelected(key: any): boolean {
+    return this.widget?.isRowSelected(key);
+  }
+
+  @Method()
+  keyOf(obj: any): any {
+    return this.widget?.keyOf(obj);
+  }
+
+  @Method()
+  navigateToRow(key: any): void {
+    return this.widget?.navigateToRow(key);
+  }
+
+  @Method()
+  pageCount(): number {
+    return this.widget?.pageCount();
+  }
+
+  @Method()
+  pageIndex(
+    newIndex: undefined | number,
+  ): Promise<void> & JQueryPromise<void> | number {
+    return this.widget?.pageIndex(newIndex);
+  }
+
+  @Method()
+  pageSize(value: number): void {
+    return this.widget?.pageSize(value);
+  }
+
+  @Method()
+  refresh(
+    changesOnly: undefined | boolean,
+  ): Promise<void> & JQueryPromise<void> {
+    return this.widget?.refresh(changesOnly);
+  }
+
+  @Method()
+  repaintRows(rowIndexes: number[]): void {
+    return this.widget?.repaintRows(rowIndexes);
+  }
+
+  @Method()
+  saveEditData(): Promise<void> & JQueryPromise<void> {
+    return this.widget?.saveEditData();
+  }
+
+  @Method()
+  searchByText(text: string): void {
+    return this.widget?.searchByText(text);
+  }
+
+  @Method()
+  selectAll(): Promise<void> & JQueryPromise<void> {
+    return this.widget?.selectAll();
+  }
+
+  @Method()
+  selectRows(
+    keys: any[], preserve: boolean,
+  ): Promise<any> & JQueryPromise<any> {
+    return this.widget?.selectRows(keys, preserve);
+  }
+
+  @Method()
+  selectRowsByIndexes(indexes: number[]): Promise<any> & JQueryPromise<any> {
+    return this.widget?.selectRowsByIndexes(indexes);
+  }
+
+  @Method()
+  showColumnChooser(): void {
+    return this.widget?.showColumnChooser();
+  }
+
+  /*
+  @Method()
+  state(state: any): any {
+    return this.widget?.state();
+  } */
```

Comment:
Why is state commented out ?

Reply:
Because JSXComponent has state field. I told about this problems to renovators.

Reply:
OK.